integration-tests/bats/cherry-pick.bats

```diff
@@ -0,0 +1,439 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+    skip_nbf_dolt_1
+
+    dolt sql -q "CREATE TABLE test(pk BIGINT PRIMARY KEY, v varchar(10))"
+    dolt add -A
+    dolt commit -m "Created table"
+    dolt checkout -b branch1
+    dolt sql -q "INSERT INTO test VALUES (1, 'a')"
+    dolt add -A
+    dolt commit -m "Inserted 1"
+    dolt sql -q "INSERT INTO test VALUES (2, 'b')"
+    dolt add -A
+    dolt commit -m "Inserted 2"
+    dolt sql -q "INSERT INTO test VALUES (3, 'c')"
+    dolt add -A
+    dolt commit -m "Inserted 3"
+}
+
+teardown() {
+    assert_feature_version
+    teardown_common
+}
+
```

Comment:
We should add a test for trying to cherry pick a merge commit. We should follow git's lead and error out if a merge commit is specified, since we don't know which of the two parents to use for calculating the diff. Git provides an option in the CLI to select which parent for a merge commit, but we can just mention that we don't support cherry-picking merge commits yet in the CLI docs somewhere for now.